ui/components/multichain/import-nfts-modal/import-nfts-modal.js

```diff
@@ -77,13 +97,24 @@ export const ImportNftsModal = ({ onClose }) => {
   const [nftAddFailed, setNftAddFailed] = useState(false);
   const trackEvent = useContext(MetaMetricsContext);
 
+  const [actionMode, setActionMode] = useState(ACTION_MODES.IMPORT_NFT);
+
+  const [selectedNetworkForCustomImport, setSelectedNetworkForCustomImport] =
+    useState(null);
+
+  const allNetworks = useSelector(getNetworkConfigurationsByChainId);
+  const networkConfigurations = useSelector(getNetworkConfigurationsByChainId);
+  const networkClientId = useSelector(getSelectedNetworkClientId);
+
   const [nftAddressValidationError, setNftAddressValidationError] =
     useState(null);
   const [duplicateTokenIdError, setDuplicateTokenIdError] = useState(null);
 
   const handleAddNft = async () => {
     try {
-      await dispatch(addNftVerifyOwnership(nftAddress, tokenId));
+      await dispatch(
+        addNftVerifyOwnership(nftAddress, tokenId, networkClientId),
```

Comment:
I think this will work today as long you are on a network where you actually own the NFT;
I have a PR on core that updates the NFTController to make the networkClientId mandatory and work properly with multichain Import; i assume we will hold this PR until we get that one first 🙏

Reply:
Sorry, just to clarify: are you suggesting that we not merge this PR until the NFTController upgrade has been released?

I guess I don't see how explicitly passing an optional value to `addNftVerifyOwnership` method cause us to need to block this PR. Am I missing something? The way I see it, this change will probably make the upgrade easier.

Reply:
Coming back to this; after looking again into the controllers, i think the bug is coming from the UI; this line https://github.com/MetaMask/metamask-extension/pull/32020/files#diff-110fd8fa49f21f99c6fc990bd47dc53fe1b4679ad6a4f052e6996012dd1f5d1fR107 is returning the selected network client ID rather than the network client ID of the network that the user chose on the form 🙏

Reply:
https://github.com/MetaMask/metamask-extension/pull/32020/commits/d985f0c338f9c6644491e4de20f8269192ba2945

Will add some tests on this as well